include/cograph_rec/Part.hpp

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <vector>
+#include <networkit/base/Algorithm.hpp>
+#include <networkit/components/ConnectedComponents.hpp>
+#include <networkit/graph/Graph.hpp>
+
+namespace Koala {
+    class element;
+
+    class part {
+    public:
+        element *first, *last, *pivot;
+        part *previous, *next, *division;

```

Comment:
Is the handwritten list necessary, or could it be substituted with std::list, std::vector or anything like that without sacrificing theoretical complexity?